File: app/src/main/java/stock/com/ui/watch_list/WatchListAdapter.java
package stock.com.ui.watch_list;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Filter;
import android.widget.Filterable;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.chauthai.swipereveallayout.ViewBinderHelper;
import stock.com.R;
import stock.com.ui.pojo.StockPojo;

import java.util.ArrayList;

public class WatchListAdapter extends RecyclerView.Adapter<ViewHolder> implements Filterable {

    private ArrayList<StockPojo.Stock> searchList=new ArrayList<>();
    private ArrayList<StockPojo.Stock> list;
    private Context context;
    private ViewBinderHelper viewBinderHelper = new ViewBinderHelper();
    private WatchListActivity activity;

    public WatchListAdapter(ArrayList<StockPojo.Stock> list,Context context,WatchListActivity activity) {
        this.context=context;
        this.searchList=list;
        this.list=list;
        this.activity=activity;
    }
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view= LayoutInflater.from(context).inflate(R.layout.row_view_watch_list,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
        viewBinderHelper.bind(holder.swipeRevealLayout, ""+position);
        holder.tv_company_name.setText(searchList.get(position).getSymbol());
        holder.tv_sector.setText(searchList.get(position).getSector());
        holder.tv_change_percentage.setText(searchList.get(position).getChangePercent());
        holder.tv_market_open.setText(searchList.get(position).getMarketopen());

        holder.img_btn_remove.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(activity!= null){
                    activity.callApiRemoveWatch(searchList.get(position).getId());
                }
            }
        });
        if(!searchList.get(position).getImage().equals("")){
            Glide.with(context).load(searchList.get(position).getImage()).into(holder.imageView);
        }

    }

    @Override
    public int getItemCount() {
        return searchList.size();
    }
    @Override
    public Filter getFilter() {
        return new Filter() {
            @Override
            protected FilterResults performFiltering(CharSequence charSequence) {
                String charString = charSequence.toString();
                if (charString.isEmpty()) {
                    searchList = list;
                } else {
                    ArrayList<StockPojo.Stock> filteredList = new ArrayList<>();
                    for (StockPojo.Stock row : list) {
                        // name match condition. this might differ depending on your requirement
                        // here we are looking for name or phone number match
                        //if (row.getFullname().toLowerCase().contains(charString.toLowerCase())) {
                        if (row.getCompanyName().toLowerCase().contains(charString.toLowerCase())||row.getSymbol().toLowerCase().contains(charString.toLowerCase())) {
                            filteredList.add(row);
                        }
                    }
                    searchList = filteredList;
                }
                FilterResults filterResults = new FilterResults();
                filterResults.values = searchList;
                return filterResults;
            }
            @Override
            protected void publishResults(CharSequence charSequence, FilterResults filterResults) {
                searchList = (ArrayList<StockPojo.Stock>) filterResults.values;
                notifyDataSetChanged();
            }
        };
    }

}

File: app/src/main/java/stock/com/ui/watch_list/WatchListAdapter.kt
package stock.com.ui.watch_list

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_match.view.*
import kotlinx.android.synthetic.main.row_view_slide_menu.view.*
import stock.com.ui.support.SupportActivity
import stock.com.ui.watch_list.WatchListActivity
import com.chauthai.swipereveallayout.ViewBinderHelper

import android.os.Bundle
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.row_latest_news.view.*
import kotlinx.android.synthetic.main.row_view_featured_contest.view.*

import kotlinx.android.synthetic.main.row_view_watch_list.view.*
import stock.com.R
import stock.com.ui.pojo.StockPojo
import stock.com.utils.AppDelegate


class WatchListAdapter(val mContext: Context, val mContest: List<StockPojo.Stock>,val activity: WatchListActivity):
    RecyclerView.Adapter<WatchListAdapter.WatchListHolder>() {
    // This object helps you save/restore the open/close state of each view
    private val viewBinderHelper = ViewBinderHelper()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WatchListHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_view_watch_list, parent, false)
        return WatchListHolder(view)
    }
    override fun onBindViewHolder(holder: WatchListHolder, position: Int) {
       // holder.itemView. tv_title_menu.setText(mContest.get(position));
         viewBinderHelper.bind(holder.itemView.swipeRevealLayout, ""+position);
         holder.itemView.tv_company_name.setText(mContest.get(position).symbol);
         holder.itemView.tv_sector.setText(mContest.get(position).sector);
         holder.itemView.tv_change_percentage.setText(mContest.get(position).changePercent);
         holder.itemView.tv_market_open.setText(mContest.get(position).marketopen);


        holder.itemView.img_btn_remove.setOnClickListener {
            if(activity!= null){
                activity.callApiRemoveWatch(mContest.get(position).id)
            }
        }

        if(!mContest.get(position).image.equals("")){
            Glide.with(mContext).load(mContest.get(position).image).into(holder.itemView.imageView)
        }

     }
    override fun getItemCount(): Int {
        return mContest.size;
    }
    inner class WatchListHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {


    }


}

